engine/access/rpc/backend/backend_scripts.go

```diff
@@ -205,3 +216,49 @@ func (b *backendScripts) tryExecuteScript(ctx context.Context, executorAddress s
 	}
 	return execResp.GetValue(), nil
 }
+
+func (b *backendScripts) tryExecuteScriptOnArchiveNode(ctx context.Context, executorAddress string, blockID flow.Identifier,
+	script []byte,
+	arguments [][]byte) ([]byte, error) {
+	req := &access.ExecuteScriptAtBlockIDRequest{
+		BlockId:   blockID[:],
+		Script:    script,
+		Arguments: arguments,
+	}
+	port, err := findPortFromAddress(executorAddress)
```

Comment:
This will be called on every script execution. Given the RN list is static, could we compute and cache the results in the constructor?

Reply:
alternatively, we could refactor the `GetAccessAPIClient` methods so the special handling to swap the libp2p port with the grpc port is only called for collection nodes. that way RNs just need to pass the original address string

Reply:
I still think you should do this. It's only a few extra lines of code to avoid the string manipulation on every script execution. On mainnet, that's 50-200 req/sec per node.

Reply:
I've made it part of the constructor as we have two active ports in the Archive node setup now, so it's more flexible